python/STS_ENABLED_IN_REGIONS/STS_ENABLE_IN_REGIONS.py

```diff
@@ -0,0 +1,425 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+'''
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  STS_ENABLE_IN_REGIONS

```

Comment:
We must avoid cross region call in AWS Config. It is best you check that a particular region as STS enabled.

Reply:
**Questions:**
- When the config rule runs by default it should run in the us-east-1 region.
  Is that correct?
- When the lambda config rule is executed it tests the connectivity to each endpoint.
  Is that considered to be cross region at that point?
- Should the code just be checking if STS is active in the region where the code is running?

Reply:
- It is possible to run a config rule, in only 1 region. The rule are independant. The goal is to decouple regions.
- As soon as you do, it makes cross region
```
session = boto3.Session(region_name=region_name)
endpoint_url = f"{sts}{region_name}{domain}"
client = session.client("sts", endpoint_url=endpoint_url)
```
- Yes, if there is no way to get all the data from a single region.